### PR TITLE
ABX-4225 Add name alias to mobile wallet list

### PR DESCRIPTION
**Pull request by desnor**
* update Wallet and Balance screen to provide input to
save and show account name property for each account
* update CreateAccount and WalletList to add new wallet to front of
wallet list instead of reversing in each render WalletList call

### CODE REVIEW

File: src/services/wallet_persistance.ts [hunk 1]
```diff
@@ -2,14 +2,12 @@ import { AsyncStorage } from 'react-native'
...
+export async function addNewWallet(publicKey: string, encryptedPrivateKey: string, accountName: string): Promise<Wallet[]> {
+  const walletEntry = { publicKey, encryptedPrivateKey, accountName }
 
-  return retrieveWallets()
-    .then((wallets: Wallet[]) => {
-      const newWalletList = [walletEntry].concat(wallets || [])
-      return saveWallets(newWalletList)
-    })
+  const wallets = await retrieveWallets()
+  const newWalletList = [walletEntry].concat(wallets)
```
Comment (Carl-Foster) [inline src/services/wallet_persistance.ts]: But `[walletEntry, ...wallets]`

RESTY

File: src/CreateAccount.tsx [hunk 2]
```diff
@@ -1,33 +1,56 @@
 import React from 'react';
 import { Alert, TouchableOpacity, ScrollView, StyleSheet, TextInput, Text, View } from 'react-native'
+import { connect, MapDispatchToProps, MapStateToProps } from 'react-redux'
+let StellarBase = require('js-kinesis-sdk')
```
Comment (Carl-Foster) [inline src/CreateAccount.tsx]: We should be importing this now since we have the types (I know it's just been moved)

File: src/CreateAccount.tsx [hunk 3]
```diff
@@ -37,231 +60,166 @@ const mapDispatchToProps: MapDispatchToProps<DispatchProps, {}> = (dispatch, own
...
-        { cancelable: false }
-      )
-      return false
-    }
-    let validPassword = await this.verifyPassword(this.state.password, this.state.passwordVerify, 'import')
-    if (validPassword) {
-      this.addNewWallet(this.state.publicKey, this.state.privateKey, this.state.password)
-      return true
-    }
+  public setAlert = async ({ header, message }: { header: string , message: string }): Promise<boolean> => {
```
Comment (Carl-Foster) [inline src/CreateAccount.tsx]: This doesn't look like it needs to be async, and neither does the validateCondition. Would be a bit cleaner without all the Promise.all

**Review by Carl-Foster — CHANGES_REQUESTED** [review]
Maybe just take out the async for some of them to save some overhead. Not a big deal but would clean it up a bit.

File: src/services/wallet_persistance.ts (hunk 1 above)
Comment (desnor) [reply src/services/wallet_persistance.ts]: :+1:

File: src/CreateAccount.tsx (hunk 3 above)
Comment (desnor) [reply src/CreateAccount.tsx]: I think I meant to call `await Alert.alert(....` as it previously seemed to wrap any Alert calls inside async functions. I don't know whether it's necessary, maybe it helps for dealing with the alert dismissal at that point. I'll look into it and take out the async if it doesn't make any difference to how it behaves.

File: src/services/wallet_persistance.ts (hunk 1 above)
Comment (desnor) [reply src/services/wallet_persistance.ts]: hang on ...SPREADY

Comment (Carl-Foster) [reply src/services/wallet_persistance.ts]: Oops true

**Review by Carl-Foster — APPROVED** [review]

**Pull request merged**